core/src/main/scala/org/bitcoins/core/hd/BIP32Path.scala

```diff
@@ -144,6 +147,21 @@ object BIP32Path {
 
     BIP32PathImpl(path)
   }
+
+  def fromBytes(bytes: ByteVector): BIP32Path = {
+    require(bytes.size % 4 == 0, "ByteVector is not suited for KeyPath")
+
+    val parts = bytes.toIterable.grouped(4).toVector
+
+    val path = parts.map { part =>
+      val uInt32: UInt32 = UInt32.fromBytes(ByteVector(part))
```

Comment:
if you remove the `.toIterable` on line 154 and remove change `ByteVector(part)` -> `part` this should be a bit cleaner. Here is what i have for this method

```scala
  def fromBytes(bytes: ByteVector): BIP32Path = {
    require(bytes.size % 4 == 0, s"ByteVector is not suited for KeyPath, got=${bytes.length}")

    val parts: Vector[ByteVector] = bytes.grouped(4).toVector

    val path = parts.map { part =>
      val uInt32: UInt32 = UInt32.fromBytes(part)
      val hardened = uInt32 > ExtKey.hardenedIdx
      val index = if (hardened) uInt32 - ExtKey.hardenedIdx else uInt32
      BIP32Node(index.toInt, hardened)
    }

    BIP32Path(path)
  }
```